prow/tide/tide.go

```diff
@@ -1194,6 +1218,45 @@ func (pr *PullRequest) logFields() logrus.Fields {
 	}
 }
 
+func (pr *PullRequest) approverTags(tag string) string {
+	if len(pr.Reviews.Nodes) == 0 {
+		return ""
+	}
+
+	approverSet := map[string]bool{}
+	for _, review := range pr.Reviews.Nodes {
+		approverSet[review.Author.String()] = true
+	}
+
+	approvers := make([]string, len(approverSet))
+	i := 0
+	for approver := range approverSet {
+		approvers[i] = approver
+		i += 1
+	}
+	sort.Strings(approvers)
+
+	lines := make([]string, len(approvers)+1)
+	for i, approver := range approvers {
+		lines[i] = fmt.Sprintf("%s: %s", tag, approver)
+	}
+	lines[len(approvers)] = ""
+	return strings.Join(lines, "\n")
+}
+
+func (author *Author) String() string {
```

Comment:
I'm not a fan of having three different output formats possible here. I'd like to have `${author.User.Name} <${author.User.Email}>` in all cases.

In the case that author.User.Name is nil, then use author.Login. In the case that author.User.Email is nil, then use `${author.Login}@users.noreply.github.com`

That way, the output is consistent for the git trailer, if we're using a standard field.

Reply:
> In the case that `author.User.Name` is `nil`, then use `author.Login`. In the case that `author.User.Email` is `nil`, then use `${author.Login}@users.noreply.github.com`

There are four cases (name set/unset * email set/unset, see the unit tests).  Are you suggesting:

* `alice <alice@users.noreply.github.com>` (empty name and email)
* `Bob <bob@users.noreply.github.com>` (name set, empty email)
* `Carol <c@example.com>` (both name and email set)
* `dan <d@example.com>` (empty name, email set)

Reply:
This sgtm

Reply:
Yeah, that works for me 👍

Reply:
Ok, I've pushed 6f8a4f4 -> 765a4db to rebase onto master, and switch to the "login as a fallback for both name and email" approach.  One caveat is that the `@users.noreply.github.com` suffix assumes `github.com`, and won't actually work if some crazy folks are using Tide with an enterprise GitHub installation (I think).  I don't mind assuming that, but thought I'd point out the possibility.